PSEUDO API 1.md

```diff
@@ -0,0 +1,197 @@
+
+## XR
+
+	interface XR {
+		Promise<sequence<XRDisplay>> getDisplays();
+	};
+
+
+## XRDisplay
+
+	interface XRDisplay : EventTarget {
+		readonly attribute DOMString displayName;
+		readonly attribute boolean isExternal;
+
+		Promise<boolean> supportsSession(XRSessionCreateParametersInit parameters);
+		Promise<XRSession> requestSession(XRSessionCreateParametersInit parameters);
+
+		attribute EventHandler ondeactivate;
+	};
+
+### Todo
+
+- calibration
+- orientation reset
+
+## XRSession
+
+	interface XRSession : EventTarget {
+		readonly attribute XRDisplay display;
+		readonly attribute XRSessionCreateParameters createParameters;
+
+		attribute double depthNear;
+		attribute double depthFar;
+
+		attribute XRLayer layer;
+		attribute Reality reality; // Defaults to an empty (VR) reality
+
+		Promise<sequence <Reality>> getRealities(); // Always returns at least the default empty reality
+		Promise<boolean> requestRealityChange(Reality reality); // resolves true if the request is accepted
+
+		Promise<XRFrameOfReference> requestFrameOfReference(XRFrameOfReferenceType type);
+
+		long requestFrame(XRFrameRequestCallback callback);
+		void cancelFrame(long handle);
+
+		long addAnchor(XRAnchor anchor);
+		void removeAnchor(long id);
+
+		Promise<void> endSession();
+
+		attribute EventHandler onblur;
+		attribute EventHandler onfocus;
+		attribute EventHandler onresetpose;
+		attribute EventHandler onended;
+		attribute EventHandler onrealitychanged;
+	};
+
+## Reality
+
+	interface Reality : EventTarget {
+		readonly atttribute DOMString realityName;
+		readonly attribute XRCoordinates stageLocation;
+		readonly attribute isPassthrough; // True if the Reality is a view of the outside world, not a fully VR
+
+		readonly attribute boolean hasPointCloud;
```

Comment:
I think we need more or different then "hasPointCloud".  Tango has a depth point cloud (dense, pretty direct representation of the depth sensor).  ARKit has a sparse set of tracked points.  Other systems may have others.

Probably want some mix of trackedPoints, depthPoints, mapMesh.